api/v1beta1/remotesecret_types.go

```diff
@@ -36,16 +36,45 @@ type RemoteSecretSpec struct {
 }
 
 type RemoteSecretTarget struct {
+	// Secret contains the overriden definitions of the secret specific to this target.
+	// +kubebuilder:validation:Optional
+	Secret *SecretOverride `json:"secret,omitempty"`
 	// Namespace is the name of the target namespace to which to deploy.
-	Namespace string `json:"namespace,omitempty"`
+	Namespace string `json:"namespace"`
```

Comment:
I believe `namespace` has to be specified and we don't really expect it to be empty. So we can be pretty sure this is never empty in the existing objects in the cluster. At the same time, this will prevent users from trying to submit a target with an empty or unspecified namespace, which is what we what we now expect only implicitly.